channeldb/control_tower.go

```diff
@@ -71,12 +73,21 @@ func NewPaymentControl(db *DB) ControlTower {
 	}
 }
 
-// ClearForTakeoff checks that we don't already have an InFlight or Completed
-// payment identified by the same payment hash.
-func (p *paymentControl) ClearForTakeoff(htlc *lnwire.UpdateAddHTLC) error {
+// InitPayment checks or records the given CreationInfo with the DB, making
+// sure it does not already exist as an in-flight payment. Then this method
+// returns successfully, the payment is guranteeed to be in the InFlight state.
+func (p *paymentControl) InitPayment(paymentHash lntypes.Hash,
+	info *CreationInfo) error {
+
+	var b bytes.Buffer
+	if err := serializeCreationInfo(&b, info); err != nil {
+		return err
+	}
+	infoBytes := b.Bytes()
+
 	var takeoffErr error
 	err := p.db.Batch(func(tx *bbolt.Tx) error {
-		bucket, err := fetchPaymentBucket(tx, htlc.PaymentHash)
+		bucket, err := fetchPaymentBucket(tx, paymentHash)
```

Comment:
`takeoffErr` needs to be set to nil at the top each time.

Reply:
it is nilled at line 100